src/primitives/transaction.h

```diff
@@ -118,10 +118,8 @@ class CTxIn
         READWRITE(nSequence);
     }
 
-    bool IsFinal() const
-    {
-        return (nSequence == std::numeric_limits<uint32_t>::max());
-    }
+    bool IsFinal() const { return (nSequence == std::numeric_limits<uint32_t>::max()); }
+    bool IsNull() const { return prevout.IsNull() || scriptSig.empty(); }
```

Comment:
(040600e2c1aa5d7f7b1c1e66cea088dc698caa77)
`IsNull()` is used to replace`== CTxIn()`, so this  should be
```suggestion
    bool IsNull() const { return prevout.IsNull() && scriptSig.empty() && IsFinal(); }
```

Otherwise, for example, you would get `IsNull() == true` for any zerocoin spend (v<3) input (prevout is null but scriptSig is not).

Reply:
yeah, it's a different behavior. We are not accepting zc spends in the tier two area (never did) but this is a primitive method.